flixel/addons/display/FlxClothSprite.hx

```diff
@@ -0,0 +1,533 @@
+				_matrix.translate( -origin.x, -origin.y);
+				_matrix.scale(scale.x, scale.y);
+				
+				if (bakedRotationAngle <= 0)
+				{
+					updateTrig();
+					
+					if (angle != 0)
+					{
+						_matrix.rotateWithTrig(_cosAngle, _sinAngle);
+					}
+				}
+				
+				_point.add(origin.x, origin.y);
+				if (isPixelPerfectRender(camera))
+				{
+					_point.floor();
+				}
+				
+				// Create a temporary frame and draw mesh's bitmapData
+				var frameMesh:FlxFrame = new FlxFrame(FlxGraphic.fromBitmapData(_meshPixels, true, null, false));
+				frameMesh.frame = new FlxRect(0, 0, _meshPixels.width, _meshPixels.height);
+				
+				_matrix.translate(_point.x + _drawOffset.x, _point.y + _drawOffset.y);
+				camera.drawPixels(frameMesh, framePixels, _matrix, cr, cg, cb, alpha, blend, antialiasing);
+			}
+			
+			#if !FLX_NO_DEBUG
+			FlxBasic.visibleCount++;
+			#end
+		}
+		
+		#if !FLX_NO_DEBUG
+		if (FlxG.debugger.drawDebug)
+		{
+			drawDebug();
+		}
+		#end
+	}
+	
+	#if !FLX_NO_DEBUG	
+	override public function drawDebugOnCamera(camera:FlxCamera):Void 
+	{
+		if (!camera.visible || !camera.exists || !isOnScreen(camera))
+		{
+			return;
+		}
+		
+		var rect = getBoundingBox(camera);
+		
+		// Find the color to use
+		var color:Null<Int> = debugBoundingBoxColor;
+		if (color == null)
+		{
+			if (allowCollisions != FlxObject.NONE)
+			{
+				color = immovable ? FlxColor.GREEN : FlxColor.RED;
+			}
+			else
+			{
+				color = FlxColor.BLUE;
+			}
+		}
+		
+		//fill static graphics object with square shape
+		var gfx:Graphics = beginDrawDebug(camera);
+		gfx.lineStyle(1, color, 0.5);
+		gfx.drawRect(rect.x, rect.y, rect.width, rect.height);
+		
+		//draw meshes and rect of pixels _meshPixels
+		gfx.lineStyle(1, FlxColor.CYAN, 0.5);
+		gfx.drawRect(rect.x + _drawOffset.x, rect.y + _drawOffset.y, _meshPixels.rect.width, _meshPixels.rect.height);
+		for (p in points) 
+		{
+			gfx.drawCircle(rect.x + p.x, rect.y + p.y, 2);
+		}
+		for (s in constraints) 
+		{
+			gfx.moveTo(rect.x + s.p0.x, rect.y + s.p0.y);
+			gfx.lineTo(rect.x + s.p1.x, rect.y + s.p1.y);
+		}
+		endDrawDebug(camera);
+	}
+	#end
+	
+	/**
+	 * Sets mesh points and constraints.
+	 * 
+	 * @param	columns				Number of columns of the created mesh.
+	 * @param	rows				Number of rows of the created mesh.
+	 * @param   meshPixelsWidth		Optional, specify the width of the bitmapData where the mesh will be drawn.
+	 * 								It uses frameWidth by default, but when mesh state is bigger, a new BitmapData is created.
+	 * @param   meshPixelsHeight	Optional, specify the height of the bitmapData where the mesh will be drawn.
+	 * 								It uses frameHeight by default, but when mesh state is bigger, a new BitmapData is created.
+	 * @param   pinned				Indices of pinned points that are not affected by wind or velocity.
+	 */
+	public function setMesh(?columns:Int = 0, ?rows:Int = 0, ?meshPixelsWidth:Int = 0, ?meshPixelsHeight:Int = 0, ?pinned:Array<Int> = null):Void
```

Comment:
I don't think there's a reason for any of the `Int`s here to have a `?` ([making them `Null<Int>`](http://stackoverflow.com/questions/31914614/is-question-mark-operator-in-haxe-really-necessary/31916466#31916466)).